heat/core/relational.py

```diff
@@ -165,6 +166,9 @@ def equal(x: Union[DNDarray, float, int], y: Union[DNDarray, float, int]) -> boo
                 y = y.balance()
 
     result_type = types.result_type(x, y)
+    is_mps = x.larray.is_mps or y.larray.is_mps
+    if is_mps and result_type is types.float64:
+        result_type = types.float32
```

Comment:
Instead of checking every time after calling ```types.result_type```, the check could be done inside ```types.result_type()```. This would save a lot of extra if statements and less chance of possibly forgetting to add that.